x/evm/types/params.go

```diff
@@ -186,7 +197,74 @@ func validateEIP712AllowedMsgs(i interface{}) error {
 	return nil
 }
 
+func validateEnabledPrecompiles(enabledPrecompiles []string) error {
+	for _, addr := range enabledPrecompiles {
+		if !common.IsHexAddress(addr) {
+			return fmt.Errorf("invalid hex address: %v in enabled precompiles list", addr)
+		}
+	}
+
+	addrs := make([]common.Address, len(enabledPrecompiles))
+	for i, precompile := range enabledPrecompiles {
+		addrs[i] = common.HexToAddress(precompile)
+	}
+
+	if err := checkIfSortedInBytesRepr(addrs); err != nil {
+		return fmt.Errorf("enabled precompiles are not sorted: %v", err)
+	}
+
+	if err := checkIfUniqueInBytesRepr(addrs); err != nil {
+		return fmt.Errorf("enabled precompiles are not unique: %v", err)
+	}
+
+	return nil
+}
+
+func checkIfSortedInBytesRepr(addrs []common.Address) error {
```

Comment:
nit: `validate` might be more consistent naming here and below than `checkIf`

Reply:
agree, `validateSortingInBytesRepr` and `validateUniquenessInBytesRepr` sounds better

Fixed: https://github.com/Kava-Labs/ethermint/pull/55/commits/27126aac5cd4e43252f2baa8ce9e8b198ba990a2